satellite/admin/user.go

```diff
@@ -166,3 +174,92 @@ func (server *Server) userInfo(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	_, _ = w.Write(data) // nothing to do with the error response, probably the client requesting disappeared
 }
+
+func (server *Server) addCoupon(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to read body: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	var input struct {
+		UserID      uuid.UUID `json:"userId"`
+		Duration    int       `json:"duration"`
+		Amount      int64     `json:"amount"`
+		Description string    `json:"description"`
+	}
+
+	err = json.Unmarshal(body, &input)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to unmarshal request: %v", err), http.StatusBadRequest)
+		return
+	}
+	switch {
+	case input.Duration == 0:
+		http.Error(w, "Duration is not set", http.StatusBadRequest)
+		return
+	case input.Amount == 0:
+		http.Error(w, "Amount is not set", http.StatusBadRequest)
+		return
+	case input.UserID.IsZero():
+		http.Error(w, "UserID is not set", http.StatusBadRequest)
+		return
```

Comment:
The description is displayed in the Deposit & Billing History. It is confusing when there is no description displayed. So perhaps, we should make it mandatory here.

Reply:
done